IPython/core/interactiveshell.py

```diff
@@ -208,6 +208,16 @@ class ExecutionResult(object):
     def success(self):
         return (self.error_before_exec is None) and (self.error_in_exec is None)
 
+    def raise_error(self):
+        """Reraises error if `success` is `True`, otherwise does nothing"""
```

Comment:
Shouldn't that be if `success` is *False*? Success is True when there isn't an error.